gcc/rust/resolve/rust-ast-resolve-pattern.h

```diff
@@ -45,6 +45,30 @@ struct PatternBinding
   {}
 };
 
+// Info that gets stored in the map. Helps us detect if two bindings to the same
+// identifier have different mutability or ref states.
+struct BindingInfo
+{
+  bool is_mut;
+  bool is_ref;
+  Location locus;
+
+  BindingInfo (bool is_mut, bool is_ref, Location locus)
+    : is_mut (is_mut), is_ref (is_ref), locus (locus)
+  {}
+  BindingInfo (){};
```

Comment:
brilliant this i think its just missing the copy constructor and assignment operator which will avoid some niggly issues